lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -362,10 +409,41 @@ private TaskStateEvent GetTaskErrorEventByExceptionType(IFailedTask failedTask)
             {
                 return TaskStateEvent.FailedTaskCommunicationError;
             }
-            else
+            if (exception is IMRUTaskSystemException)
             {
                 return TaskStateEvent.FailedTaskSystemError;
             }
+
+            // special case for communication error during group communication initialization
+            if (exception is TaskClientCodeException)
+            {
+                {
+                    // try extract cause and check whether it is InjectionException for GroupCommClient
+                    if (exception.InnerException != null &&
+                        exception.InnerException is InjectionException &&
+                        exception.InnerException.Message.Contains("GroupCommClient"))
+                    {
+                        Logger.Log(Level.Info, "GetTaskErrorEventByExceptionType:FailedTaskCommunicationError with task id {0}", failedTask.Id);
+                        return TaskStateEvent.FailedTaskCommunicationError;
+                    }
+                }
+            }
+
+            if (exception is AggregateException)
```

Comment:
Why do we need to treat this as separate? Why is this not captured by TaksHost and wrapped properly in IMRU exception and detected above?